modules/virtualnetwork/main.tf

```diff
@@ -155,10 +156,96 @@ resource "azapi_resource" "vhubconnection" {
           id = each.value.vwan_associated_routetable_resource_id != "" ? each.value.vwan_associated_routetable_resource_id : "${each.value.vwan_hub_resource_id}/hubRouteTables/defaultRouteTable"
         }
         propagatedRouteTables = {
-          ids    = local.vwan_propagated_routetables_resource_ids[each.key]
-          labels = local.vwan_propagated_routetables_labels[each.key]
+          ids    = each.value.vwan_security_configuration.secure_private_traffic == true ? local.vwan_propagated_noneroutetables_resource_ids[each.key] : local.vwan_propagated_routetables_resource_ids[each.key]
+          labels = each.value.vwan_security_configuration.secure_private_traffic == true ? ["none"] : local.vwan_propagated_routetables_labels[each.key]
         }
       }
     }
   })
 }
+
+# azapi_update_resource.vhubdefaultroutetableinternettraffic creates public_traffic route which associated vnets then learn
+# with a next hop route for the internet as the vhub's azure firewall
+resource "azapi_update_resource" "vhubdefaultroutetableinternettraffic" {
```

Comment:
I don't think we should update hub route tables using this module.

1. This does not scale, with 100s of instances of this module the number of API calls will cause rate limiting.
2. I do not believe this module is the right place to do this. Any hub route table config should be performed using the terraform code that deploys the vwan hub resource.

Reply:
I had similar thoughts while testing.

Reply:
Thanks - I'm happy to remove this but if you want to then that would be helpful.